src/commands.cpp

```diff
@@ -4885,12 +4925,13 @@ bool CommandSetPH::procresult(Result r)
     }
 
     Node *n = client->nodebyhandle(h);
-    if (n)
-    {
-        n->setpubliclink(ph, time(nullptr), ets, false);
-        n->changed.publiclink = true;
-        client->notifynode(n);
-    }
+    assert(n->plink);
```

Comment:
If the AP has not been received yet, then the `n` won't have a `plink`, right?

Reply:
When there's an actionpacket involved, which is signalled by the presence of the seqtag, we don't call the command's procresult() until we have received both the command response, and the actionpacket itself (and we have processed that actionpacket - as well as any others that follow that actionpacket and have the same seqtag).     Whichever of the command processing or actionpacket processing that receives an unmatched seqtag, waits for the other to catch up (or reach a condition where we know there won't be a match) and when we have both (and actually, just before we process the actionpacket after), then we invoke the command's procresult().

Reply:
Oh, I thought the logic was slightly different. So the APs are always processed first, and the `procresult()` is called only once the corresponding APs with the same `seqtag` are processed, correct?
Then the assert makes perfect sense :smiley:

Reply:
Yes - the command result and the actionpackets are in a race and either could arrive first on the network.   However, regardless of that, we always match up the command and actionpacket, and the actionpackets are processed, before the command's procresult() is called.